streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1522,6 +1525,72 @@ private void maybeScheduleFollowupRebalance(final long encodedNextScheduledRebal
         }
     }
 
+    private AssignmentError validateTaskAssignment(final ApplicationState applicationState,
+                                                   final TaskAssignment taskAssignment) {
+        final Collection<KafkaStreamsAssignment> assignments = taskAssignment.assignment();
+        final Set<TaskId> activeTasksInOutput = new HashSet<>();
+        final Set<TaskId> standbyTasksInOutput = new HashSet<>();
+        for (final KafkaStreamsAssignment assignment : assignments) {
+            final Set<TaskId> tasksForAssignment = new HashSet<>();
+            for (final KafkaStreamsAssignment.AssignedTask task : assignment.assignment()) {
+                if (activeTasksInOutput.contains(task.id()) && task.type() == KafkaStreamsAssignment.AssignedTask.Type.ACTIVE) {
+                    log.error("Assignment is invalid: an active task was assigned multiple times: {}", task.id());
+                    return AssignmentError.ACTIVE_TASK_ASSIGNED_MULTIPLE_TIMES;
+                }
+
+                if (tasksForAssignment.contains(task.id())) {
+                    log.error("Assignment is invalid: both an active and standby assignment of a task were assigned to the same client: {}", task.id());
+                    return AssignmentError.ACTIVE_AND_STANDBY_TASK_ASSIGNED_TO_SAME_KAFKASTREAMS;
+                }
+
+                tasksForAssignment.add(task.id());
+                if (task.type() == KafkaStreamsAssignment.AssignedTask.Type.ACTIVE) {
+                    activeTasksInOutput.add(task.id());
+                } else {
+                    standbyTasksInOutput.add(task.id());
+                }
+            }
+        }
+
+        for (final TaskInfo task : applicationState.allTasks()) {
+            if (!task.isStateful() && standbyTasksInOutput.contains(task.id())) {
+                log.error("Assignment is invalid: a standby task was found for a stateless task: {}", task.id());
```

Comment:
Same here, would be nice to log which client it was assigned to (which means turning `standbyTasksInOutput` into a Map<TaskId, ProcessId> as well). Then:
```suggestion
                log.error("Assignment is invalid: standby task for stateless task {} was assigned to KafkaStreams client {}", task.id(), standbyTasksInOutput.get(task.id().id());
```